fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/entity/FabricEntityType.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.object.builder.v1.entity;
+
+import java.util.function.Supplier;
+import java.util.function.UnaryOperator;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.SpawnGroup;
+import net.minecraft.entity.SpawnRestriction;
+import net.minecraft.entity.attribute.DefaultAttributeContainer;
+import net.minecraft.entity.mob.MobEntity;
+import net.minecraft.world.Heightmap;
+
+import net.fabricmc.fabric.impl.object.builder.FabricEntityTypeImpl;
+
+/**
+ * General-purpose Fabric-provided extensions for {@link EntityType}.
+ */
+public interface FabricEntityType {
+	/**
+	 * General-purpose Fabric-provided extensions for {@link EntityType.Builder}.
+	 *
+	 * <p>Note: This interface is automatically implemented on {@link EntityType.Builder} via Mixin and interface injection.
+	 */
+	interface Builder<T extends Entity> {
+		/**
+		 * Sets whether the entity's velocity should always be updated.
+		 *
+		 * @param alwaysUpdateVelocity whether the entity's velocity should always be updated
+		 * @return this builder
+		 */
+		default EntityType.Builder<T> alwaysUpdateVelocity(boolean alwaysUpdateVelocity) {
+			throw new AssertionError("Implemented in Mixin");
+		}
+
+		/**
+		 * Build the entity type from the builder. Same as {@link EntityType.Builder#build(String)} but without an id.
+		 *
+		 * @return the entity type instance
+		 */
+		default EntityType<T> build() {
```

Comment:
Both need `fabric_` prefixing.

Reply:
No, this is part of the API.

Reply:
Oh is this an existing API?

Reply:
No, this is new (based off the old one). The vanilla build() method takes a string that will almost always be null for mods, this is an overload of the vanilla build().

Reply:
Yes exactly.

Reply:
Fair enough, will bear in mind 👍